api/kroxylicious-filter-api/src/main/java/io/kroxylicious/proxy/filter/FilterInvokers.java

```diff
@@ -119,6 +123,40 @@ public boolean shouldHandleRequest(ApiKeys apiKey, short apiVersion) {
     public boolean shouldHandleResponse(ApiKeys apiKey, short apiVersion) {
         return responseFilter.shouldHandleResponse(apiKey, apiVersion);
     }
+
+    }
+
+    private record SafeInvoker(FilterInvoker invoker) implements FilterInvoker {
```

Comment:
`Safe` as in `TypeSafe`? Maybe `ConditionalInvoker` or `DelegatingInvoker` the naming is not quite right for me, but this is the shape I was expecting.

Reply:
Safe as in it is safe to call `on[Request|Response]` with (key, apiVersion) combinations that are rejected by `shouldHandle[Request|Response]`. The SpecificInvoker would throw an exception if you did.

Reply:
Let's leave the name as it is. I still can't think of a better option and its not important enough to hold things up for.